fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/ExtractionFunctionDimension.java

```diff
@@ -0,0 +1,21 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.dimension.impl;
+
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.druid.model.dimension.extractionfunction.ExtractionFunction;
+
+import java.util.Optional;
+
+/**
+ * A dimension that supports extraction functions.
+ */
+public interface ExtractionFunctionDimension extends Dimension {
+
+    /**
+     * Build an extraction function model object.
+     *
+     * @return  Take the internal namespaces and construct a model object for the extraction functions.
+     */
+    Optional<ExtractionFunction> getExtractionFunction();
```

Comment:
No.  We don't want to change or complicate the Dimension interface if we can avoid it.